AdminWebsite/AdminWebsite/ClientApp/src/app/services/search.service.ts

```diff
@@ -60,34 +63,56 @@ export class SearchService {
 
     constructor(private bhClient: BHClient) {}
 
-    search(terms: Observable<string>, role: string) {
-        return terms
-            .pipe(debounceTime(500))
-            .pipe(distinctUntilChanged())
-            .pipe(
-                switchMap(term => {
-                    if (role === 'Panel Member' || role === 'Winger') {
-                        return this.searchJudiciaryEntries(term);
-                    } else {
-                        return this.searchEntries(term);
-                    }
+    participantSearch(term: string, role: string): Observable<Array<ParticipantModel>> {
+        if (role === this.judgeRole) {
+            return this.searchJudgeAccounts(term).pipe(
+                map(judges => {
+                    return judges.map(judge => {
+                        return ParticipantModel.fromJudgeResponse(judge);
+                    });
+                })
+            );
+        } else {
+            let persons$: Observable<Array<PersonResponse>>;
+            if (this.judiciaryRoles.includes(role)) {
+                persons$ = this.searchJudiciaryEntries(term);
+            } else {
+                persons$ = this.searchEntries(term);
+            }
+
+            return persons$.pipe(
+                map(persons => {
+                    return persons.map(person => {
+                        return ParticipantModel.fromPersonResponse(person);
+                    });
                 })
             );
+        }
     }
 
     searchEntries(term): Observable<Array<PersonResponse>> {
         const allResults: PersonResponse[] = [];
-        if (term.length > 2) {
+        if (term.length >= this.minimumSearchLength) {
             return this.bhClient.postPersonBySearchTerm(term);
+        } else {
+            return of(allResults);
         }
-        return of(allResults);
     }
 
     searchJudiciaryEntries(term): Observable<Array<PersonResponse>> {
         const allResults: PersonResponse[] = [];
-        if (term.length > 2) {
+        if (term.length >= this.minimumSearchLength) {
             return this.bhClient.postJudiciaryPersonBySearchTerm(term);
+        } else {
+            return of(allResults);
+        }
+    }
+
+    searchJudgeAccounts(term): Observable<Array<JudgeResponse>> {
+        if (term.length >= this.minimumSearchLength) {
```

Comment:
I would remove the check from here & just add it to the component (in the pipe use a filter)